src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -160,17 +161,17 @@ export class MockHttpSocket extends MockSocket {
     return emitEvent()
   }
 
-  public destroy(error?: Error | undefined): this {
+  public _destroy(
+    error?: Error | null,
+    callback?: (error?: Error | null) => void
+  ): void {
     // Destroy the response parser when the socket gets destroyed.
     // Normally, we shoud listen to the "close" event but it
     // can be suppressed by using the "emitClose: false" option.
     this.responseParser.free()
 
-    if (error) {
-      this.emit('error', error)
-    }
-
-    return super.destroy(error)
+    callback?.(error)
+    process.nextTick(() => this.emit('close'))
```

Comment:
This seems to be required to mimic the Node.js behaviour for the appropriate events to be emitted (see the `http-res-destroy` tests))

Reply:
Could you please link the Node.js source that we are mimicking here? That would be really helpful.